changelog.d/9581.feature

```diff
@@ -1 +1 @@
-Add get_userinfo_by_id method to ModuleApi.
+Add `get_userinfo_by_id` method to ModuleApi. `UserInfo` is a new type returned from the registration store. This also deprecates the `get_user_by_id` method on the registration store.
```

Comment:
`get_user_by_id` was never exposed outside internal code, so I don't think we need to mention this here.

```suggestion
Add `get_userinfo_by_id` method to ModuleApi.
```